test/integration_tests/long/test_similarity.py

```diff
@@ -388,3 +392,39 @@ def test_end_to_end_index_scan_should_work_correctly_on_image_dataset_qdrant(sel
 
         res_batch = execute_query_fetch_all(self.evadb, select_query)
         self.assertEqual(res_batch.frames["testsimilarityimagedataset._row_id"][0], 5)
+
+        # Cleanup
+        self.evadb.catalog().drop_index_catalog_entry("testQdrantIndexImageDataset")
+
+    def test_end_to_end_index_scan_should_work_correctly_on_image_dataset_pinecone(
+        self,
+    ):
+        # Set the env variables.
+        original_pinecone_key = os.environ.get("PINECONE_API_KEY")
+        original_pinecone_env = os.environ.get("PINECONE_ENV")
+
+        os.environ["PINECONE_API_KEY"] = "657e4fae-7208-4555-b0f2-9847dfa5b818"
+        os.environ["PINECONE_ENV"] = "gcp-starter"
+
+        # We need to always drop the index as Pinecone's free tier only supports a single current index.
+        drop_index_query = "DROP INDEX IF EXISTS testpineconeindeximagedataset;"
+        execute_query_fetch_all(self.evadb, drop_index_query)
+        create_index_query = """CREATE INDEX testpineconeindeximagedataset
+                                ON testSimilarityImageDataset (DummyFeatureExtractor(data))
+                                USING PINECONE;"""
+        execute_query_fetch_all(self.evadb, create_index_query)
+        # Sleep to ensure the pinecone records get updated as Pinecone is eventually consistent.
+        time.sleep(20)
```

Comment:
@xzdandy @jiashenC @gaurav274 @jarulraj since Pinecone is eventually consistent, we can't immediately query results after creating indexes. Hence i had to put a sleep in between here. Going through the documentation, I couldnt find a proper solution for this.

One suggestion was to look at the total_vector_count in describe_index_stats() fn https://docs.pinecone.io/docs/insert-data and wait till it gets updated but this won't work for updates. Any suggestions on this? Should we just assume that the clients will wait for the results to persiste before querying.

Reply:
I am not able to find a flush function either. For testcases, I think sleep is fine. One option is to print out a warning message when user is creating index using pinecone to indicate that the operation is eventual consistent.